Update VMRay pack to 1.1.0 and start adoption process

fseele-vmray opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
This PR updates the VMRay pack to version 1.1.0 which adds a new Playbook, new commands and enhances existing commands (see release notes for details).
It also starts the adoption process to move support from Cortex to VMRay.

- Adds the **Detonate URL - VMRay** playbook
- Adds the **vmray-upload-url** command
- Adds the **vmray-get-summary** command
- Adds the **vmray-get-sample-by-hash** command
- Updates the **vmray-get-sample** command
- Updates the **vmray-get-iocs** command


## Minimum version of Cortex XSOAR
- [X] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=13683) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=13683) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=13683) it.</sub>

marcellarichmond commented:
@mchasepan fyi

fseele-vmray commented:
@mchasepan Thanks for the review, I reverted the changes in the metadata.
Some questions regarding the validation issues:

> [IN140] - The integration vmray is currently in skipped. Please add working tests and unskip.

What does this mean?

> [BC102] - Possible backwards compatibility break, You've changed the context in the file,please undo. the command is:
> vmray-get-iocs

We only added to the context and did not remove or change any values. Is already considered to break backwards compatibility? Is there any way to get details on which fields it complains about exactly?

> [BC104] - Possible backwards compatibility break, You've changed the name of a command or its arg in the file, please undo, the command was:
> vmray-upload-sample

We removed a parameter which is not necessary anymore. I assumed it would not be a problem to remove it from the yml. What is the best way to do this? Leave it in the yml, mark it as being ignored and just don't use it in the code?

mchasepan commented:
@Shellyber can you please review Felix's [questions](https://github.com/demisto/content/pull/13683#issuecomment-879275498) above?

Shellyber reviewed (commented):
Thank you for your contribution! Looks good!
Let's schedule a call, I have some questions regarding some of the changes. Also we would like to have a demo for all the changes and the new playbook.

Shellyber commented:
@fseele-vmray Regarding your question, I'm checking in. I think that you can remove the argument from the yml file also without it breaking BC.

fseele-vmray commented on Packs/VMRay/Playbooks/playbook-VMRay-Detonate-File.yml:
```diff
@@ -1,7 +1,7 @@
 id: 79b5d8a6-2636-480c-8e1c-a3ab2e58ffb5
```
Is it possible to change the ID without breaking backwards-compatibility?

Shellyber replied:
No. If the id is the same as the name that's ok and we support it

fseele-vmray replied:
I applied the change but now `demisto-sdk upload` fails with this error:
```
FAILED UPLOADS:
╒══════════════════════════════════╤══════════╤══════════════════════════════════════════════════════════════════════════════════╕
│ NAME                             │ TYPE     │ ERROR                                                                            │
╞══════════════════════════════════╪══════════╪══════════════════════════════════════════════════════════════════════════════════╡
│ playbook-VMRay-Detonate-File.yml │ playbook │ Detonate File - VMRay (79b5d8a6-2636-480c-8e1c-a3ab2e58ffb5) already exists (33) │
╘══════════════════════════════════╧══════════╧══════════════════════════════════════════════════════════════════════════════════╛
```
But this change will not cause any problems for users?

Shellyber replied:
@fseele-vmray The playbook already exists in your XSOAR. So it says that the ID already exists. We can't upload two content items with the same id. 
It won't cause problems to users. If you delete the existing playbook in your env, you'll be able to upload this as well.

fseele-vmray commented on Packs/VMRay/ReleaseNotes/1_1_0.md:
```diff
@@ -0,0 +1,29 @@
+*Note: Support for this pack will be moving to the partner around Oct 15, 2021.*
+
+
+#### Overview
```
Done

fseele-vmray commented on Packs/VMRay/Integrations/VMRay/README.md:
```diff
@@ -1,1418 +1,1085 @@
-            "SampleID": 3902285, 
-            "FileName": "KeePass-2.41-Setup.exe", 
-            "FileSize": 3301376, 
+            "SHA1": "69df095557346b3c136db4378afd5ee7a4839dcc",
```
Yes 👍

Shellyber commented:
@fseele-vmray Let's schedule a demo for all the changes and the new playbook

evisochek commented on Packs/VMRay/Playbooks/playbook-VMRay-Detonate-URL.yml:
```diff
@@ -0,0 +1,652 @@
+    task:
+      description: check term
+      version: -1
+      name: Is analysis suspicious or malicious?
```
Please consider to organize the task for malicious and suspicious In a column and not in parallel so the path will be more clear to the user.

fseele-vmray replied:
Done in 2b657c1

evisochek commented:
Please add image for the playbook.

evisochek commented on Packs/VMRay/Playbooks/playbook-VMRay-Detonate-URL_README.md:
```diff
@@ -0,0 +1,101 @@
+
+## Playbook Image
+---
+![Detonate URL - VMRay](Insert the link to your image here)
```
Please add image

fseele-vmray replied:
Done in b775f79 and a8e1965

evisochek commented on Packs/VMRay/Playbooks/playbook-VMRay-Detonate-URL.yml:
```diff
@@ -0,0 +1,652 @@
+  description: The operation that caused the indicators.
+fromversion: 5.5.0
+tests:
+- no test
```
I see that there is VMRay-Test-URL playbook, should he test the VMRay-Detonate-URL playbook? if so please add him.

fseele-vmray replied:
Done in1faa07e1a7c353b5645af0f521e78c209fe3ac92

Shellyber commented on Packs/VMRay/Playbooks/playbook-VMRay-Detonate-File.yml:
```diff
@@ -1,7 +1,7 @@
-id: 79b5d8a6-2636-480c-8e1c-a3ab2e58ffb5
+id: Detonate File - VMRay
```
```suggestion
id: 79b5d8a6-2636-480c-8e1c-a3ab2e58ffb5
```
